### PR TITLE
Added a 'setting up keycloak' page to the developer guide.

### PR DESCRIPTION
**Pull request by dusktreader**
#### What
Added a page describing how to set up keycloak for jobbergate.

#### Why
To capture notes used to create the keycloak instance in `jobbergate-composed`.

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

**Review by fschuch — APPROVED** [review]

File: jobbergate-docs/src/developer_guide/keycloak_setup.rst [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+Though Jobbergate should be compatible with any OIDC provider (through the `Armasec package
+<https://github.com/omnivector-solutions/armasec>`_), the recommended provider is Keycloak.
+
+This document describes the procedure for setting up an already deployed Keycloak instance to work with Jobbergate.
+
+The version of Keycloak used for this guide is `19.0.2`.
+
+The examples provided here describe how to set it up for a local deployment (for example, one deployed with Docker).
+However, the instructions should be easily extensible for any deployment.
```
Comment (matheushent) [inline jobbergate-docs/src/developer_guide/keycloak_setup.rst:13]:
```suggestion
However, the instructions should be easily extensible for any deployment considering a single-node Keycloak cluster.
```

Comment (dusktreader) [reply jobbergate-docs/src/developer_guide/keycloak_setup.rst]: Resolved in commit 7d3ace4

File: jobbergate-docs/src/developer_guide/keycloak_setup.rst [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
++---------------------------------+-------------------------------------------+
+| jobbergate:job-scripts:view     | Allow to view job scripts                 |
++---------------------------------+-------------------------------------------+
+| jobbergate:job-submissions:edit | Allow to edit job submissions             |
++---------------------------------+-------------------------------------------+
+| jobbergate:job-submissions:view | Allow to view job submissions             |
++---------------------------------+-------------------------------------------+
+| license-manager:booking:edit    | Allow to edit bookings                    |
++---------------------------------+-------------------------------------------+
+| license-manager:booking:view    | Allow to view bookings                    |
```
Comment (matheushent) [inline jobbergate-docs/src/developer_guide/keycloak_setup.rst:91]: are these two really needed?

Comment (dusktreader) [reply jobbergate-docs/src/developer_guide/keycloak_setup.rst:91]: Good point. Not in the Jobbergate docs!

**Pull request merged**